### ISSUE TITLE
failing tests not alwasy shown correctly

### ISSUE THREAD

**Issue by rhaas80** [EinsteinToolkit/tests#33]
Right now the life view shows that CI is failing but the most recent job shows no failure:  
  
 

![](https://user-images.githubusercontent.com/226374/205136133-032a6136-178b-4264-bcb1-49aba2f969e1.png)

The action that fails is: https://github.com/EinsteinToolkit/tests/actions/runs/3595463643/jobs/6054986543#step:7:281

**Comment by rhaas80** [comment]
Looks like the libpygit2 issue we saw before:

```
       gcc -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -I/usr/local/include -I/opt/hostedtoolcache/Python/3.11.0/x64/include/python3.11 -c src/blob.c -o build/temp.linux-x86_64-cpython-311/src/blob.o
      In file included from src/blob.h:34,
                       from src/blob.c:30:
      src/types.h:36:2: error: #error You need a compatible libgit2 version (v0.28.x)
         36 | #error You need a compatible libgit2 version (v0.28.x)
            |  ^~~~~
      error: command '/usr/bin/gcc' failed with exit code 1
      [end of output]
```

**Comment by rhaas80** [comment]
maybe fixed by forcing Ubuntu 20.04 rather  than "latest" in workflow yaml file in 49214df Let's see if this fixes things for good. I am wondering if `ubuntu-latest` may be a different Ubuntu version depending on which runner host one ends on? Or maybe something changed in `ubuntu-lates`? In any case, specifying the exact version we want and not some unspecific `latest` will help reproduce issues and thus is the way to go anyway.

**Comment by nadinekuo** [comment]
@rhaas80 https://einsteintoolkit.github.io/tests/ currently only displays HTML pages corresponding to jobs for which the build phase has succeeded (and thus tests can be run), i.e. in the above case `logpage.py` is never reached and there are no test results to be shown for that job. Would it be an idea to let `logpage.py` fetch the workflow API and display such jobs as well, with the HTML containing something like: "Build failed" instead of the test results?
